### PR TITLE
Nonlive migration

### PR DESCRIPTION
**Pull request by tych0**
Implement `lxc move ...` for containers that aren't running. I'll do copy on Monday.

### CODE REVIEW

File: lxd/images.go [hunk 1]
```diff
@@ -547,20 +546,19 @@ func imageExport(d *Daemon, r *http.Request) Response {
...
 
-	// test compression, for content type header
-	// if unknown compression we send standard header
-	_, ext, err := detectCompression(path)
-
-	ctype := "application/x-gtar"
-	if err != nil {
-		ctype = "application/octet-stream"
-	}
-
```
Comment (stgraber) [inline lxd/images.go]: Why are we dropping setting Content-Type here?

Comment (tych0) [inline lxd/images.go]: On Fri, Mar 13, 2015 at 03:49:42PM -0700, St�phane Graber wrote:
> > @@ -547,20 +546,19 @@ func imageExport(d *Daemon, r *http.Request) Response {
> >  	path := shared.VarPath("images", hash)
> >  	filename := imgInfo.Filename
> >  
> > -	// test compression, for content type header
> > -	// if unknown compression we send standard header
> > -	_, ext, err := detectCompression(path)
> > -
> > -	ctype := "application/x-gtar"
> > -	if err != nil {
> > -		ctype = "application/octet-stream"
> > -	}
> > -
> 
> Why are we dropping setting Content-Type here?

It's autodetected by http.ServeContent, along with a few others.

File: test/basic.sh [hunk 2]
```diff
@@ -12,13 +12,18 @@ test_basic_usage() {
   # Test image export
   sum=$(lxc image info testimage | grep ^Hash | cut -d' ' -f2)
   lxc image export testimage ${LXD_DIR}/
-  [ "$sum" = "$(sha256sum ${LXD_DIR}/${sum}.tar.xz | cut -d' ' -f1)" ]
+  if [ -e "$LXD_TEST_IMAGE" ]; then
+      name=$(basename $LXD_TEST_IMAGE)
+  else
+      name=${sum}.tar.xz
+  fi
+  [ "$sum" = "$(sha256sum ${LXD_DIR}/${name} | cut -d' ' -f1)" ]
```
Comment (stgraber) [inline test/basic.sh]: Oh, oops, sorry about that.

**Comment by stgraber** [comment]
For the tests, I suspect we should just always spawn two LXDs since Serge's branch will require two LXDs to test creating containers from remote images. Since there's no reason those tests won't run under Travis (until we attempt a launch or start that is), I can't think of a case where we won't need two LXDs.

File: lxd/images.go (hunk 1 above)
Comment (stgraber) [inline lxd/images.go]: Oh, neat.

**Pull request merged**